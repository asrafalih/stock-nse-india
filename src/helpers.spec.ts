import {
    getGainersAndLosersByIndex,
    getMostActiveEquities
} from './helpers'
import { IndexEquityInfo } from './interface'

jest.setTimeout(999999)

describe('Helpers', () => {
    test('getGainersAndLosersByIndex', async () => {
        const data = await getGainersAndLosersByIndex("NIFTY 50")
        const isLosers = data.loosers.every((equityDetails:IndexEquityInfo) => equityDetails.pChange <= 0)
        const isGainers = data.gainers.every((equityDetails:IndexEquityInfo) => equityDetails.pChange > 0)
        expect(isLosers).toBeTruthy()
        expect(isGainers).toBeTruthy()
        const equityGainersLoosers = await getGainersAndLosersByIndex("all")
        expect(equityGainersLoosers.gainers.length).toBe(20)
        expect(equityGainersLoosers.loosers.length).toBe(20)

    })

    test('getMostActiveEquities', async () => {
        const data = await getMostActiveEquities("NIFTY 50")
        expect(data.byVolume[0] >= data.byVolume[1]).toBeTruthy()
        expect(data.byValue[0] >= data.byValue[1]).toBeTruthy()
    })
})
