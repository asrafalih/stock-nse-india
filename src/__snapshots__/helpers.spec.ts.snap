// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Helpers getGainersAndLosersByIndex: API Response Validation 1`] = `
Array [
  Object {
    "gainers": Array [
      "priority: number",
      "symbol: string | null",
      "identifier: string | null",
      "series: string | null",
      "open: number",
      "dayHigh: number",
      "dayLow: number",
      "lastPrice: number",
      "previousClose: number",
      "change: number",
      "pChange: number",
      "totalTradedVolume: number",
      "totalTradedValue: number",
      "lastUpdateTime: string | null",
      "yearHigh: number",
      "ffmc: number",
      "yearLow: number",
      "nearWKH: number",
      "nearWKL: number",
      "perChange365d: number",
      "date365dAgo: string | null",
      "chart365dPath: string | null",
      "date30dAgo: string | null",
      "perChange30d: number",
      "chart30dPath: string | null",
      "chartTodayPath: string | null",
      Object {
        "meta": Array [
          "symbol: string | null",
          "companyName: string | null",
          Object {
            "activeSeries": "string",
          },
          Object {
            "debtSeries": Array [],
          },
          Object {
            "tempSuspendedSeries": Array [],
          },
          "isFNOSec: boolean",
          "isCASec: boolean",
          "isSLBSec: boolean",
          "isDebtSec: boolean",
          "isSuspended: boolean",
          "isETFSec: boolean",
          "isDelisted: boolean",
          "isin: string | null",
        ],
      },
    ],
  },
  Object {
    "losers": Array [
      "priority: number",
      "symbol: string | null",
      "identifier: string | null",
      "series: string | null",
      "open: number",
      "dayHigh: number",
      "dayLow: number",
      "lastPrice: number",
      "previousClose: number",
      "change: number",
      "pChange: number",
      "totalTradedVolume: number",
      "totalTradedValue: number",
      "lastUpdateTime: string | null",
      "yearHigh: number",
      "ffmc: number",
      "yearLow: number",
      "nearWKH: number",
      "nearWKL: number",
      "perChange365d: number",
      "date365dAgo: string | null",
      "chart365dPath: string | null",
      "date30dAgo: string | null",
      "perChange30d: number",
      "chart30dPath: string | null",
      "chartTodayPath: string | null",
      Object {
        "meta": Array [
          "symbol: string | null",
          "companyName: string | null",
          "industry: string | null",
          Object {
            "activeSeries": "string",
          },
          Object {
            "debtSeries": Array [],
          },
          Object {
            "tempSuspendedSeries": "string",
          },
          "isFNOSec: boolean",
          "isCASec: boolean",
          "isSLBSec: boolean",
          "isDebtSec: boolean",
          "isSuspended: boolean",
          "isETFSec: boolean",
          "isDelisted: boolean",
          "isin: string | null",
        ],
      },
    ],
  },
]
`;

exports[`Helpers getMostActiveEquities: API Response Validation 1`] = `
Array [
  Object {
    "byVolume": Array [
      "priority: number",
      "symbol: string | null",
      "identifier: string | null",
      "open: number",
      "dayHigh: number",
      "dayLow: number",
      "lastPrice: number",
      "previousClose: number",
      "change: number",
      "pChange: number",
      "ffmc: number",
      "yearHigh: number",
      "yearLow: number",
      "totalTradedVolume: number",
      "totalTradedValue: number",
      "lastUpdateTime: string | null",
      "nearWKH: number",
      "nearWKL: number",
      "perChange365d: number",
      "date365dAgo: string | null",
      "chart365dPath: string | null",
      "date30dAgo: string | null",
      "perChange30d: number",
      "chart30dPath: string | null",
      "chartTodayPath: string | null",
    ],
  },
  Object {
    "byValue": Array [
      "priority: number",
      "symbol: string | null",
      "identifier: string | null",
      "open: number",
      "dayHigh: number",
      "dayLow: number",
      "lastPrice: number",
      "previousClose: number",
      "change: number",
      "pChange: number",
      "ffmc: number",
      "yearHigh: number",
      "yearLow: number",
      "totalTradedVolume: number",
      "totalTradedValue: number",
      "lastUpdateTime: string | null",
      "nearWKH: number",
      "nearWKL: number",
      "perChange365d: number",
      "date365dAgo: string | null",
      "chart365dPath: string | null",
      "date30dAgo: string | null",
      "perChange30d: number",
      "chart30dPath: string | null",
      "chartTodayPath: string | null",
    ],
  },
]
`;
